sql/src/main/java/io/crate/jobs/UpsertByIdContext.java

```diff
@@ -113,6 +119,7 @@ public void close() {
 
     @Override
     public void kill() {
-        throw new UnsupportedOperationException("kill is not implemented");
+        futureResult.setException(new JobKilledException());
```

Comment:
it's not a problem, because a second future.set doesn't have an effect at all, but i changed it, so the results are not processed unnecessary in this case.